src/software/ai/navigator/trajectory/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,109 @@
+#include "software/ai/navigator/trajectory/trajectory_planner.h"
+
+#include <gtest/gtest.h>
+
+#include "software/ai/navigator/obstacle/robot_navigation_obstacle_factory.h"
+#include "software/test_util/test_util.h"
+
+class TrajectoryPlannerTest : public testing::Test
```

Comment:
there would be a couple of tests that would be nice to see:

- avoiding the enemy half + centre circle
- avoiding the enemy half (no centre circle avoidance)
- a path ending in a blocked obstacle (eg. enemy half with enemy half constraint)

Reply:
I've added some sort of a test for the first two, but didn't add one for the last one since we currently don't have the implementation for moving destinations inside obstacles out of obstacles. This is something that is being tracked here #3105